### PR TITLE
Backend part of legacy IRC key management

### PR DESCRIPTION
**Pull request by nanaya**
It's also large but mostly copypasta from #10085. Also it most likely conflicts with the aforementioned PR on the route file.

### CODE REVIEW

File: app/Libraries/OsuAuthorize.php [hunk 1]
```diff
@@ -1788,6 +1788,19 @@ public function checkIsSpecialScope(?User $user): string
         return 'unauthorized';
     }
 
+    public function checkLegacyIrcKeyStore(?User $user): string
+    {
+        $this->ensureLoggedIn($user);
+        $this->ensureCleanRecord($user);
+
+        // isBot checks user primary group
+        if (!$user->isGroup(app('groups')->byIdentifier('bot')) && $user->playCount() < 100) {
```
Comment (notbakaneko) [inline app/Libraries/OsuAuthorize.php:1797]: I'm not sure if we want to change this from being primary group only? @peppy

**Review by notbakaneko — CHANGES_REQUESTED** [review]

File: app/Libraries/OsuAuthorize.php (hunk 1 above)
Comment (nanaya) [reply app/Libraries/OsuAuthorize.php:1797]: after looking at the requirement apparently you can't even be in bot group before running a bot first so this check probably doesn't matter whether or not it's primary group (the current one in old site doesn't require primary group).

Also having main account being in bot group isn't even a thing anymore anyway? (at least no new one)

Comment (notbakaneko) [reply app/Libraries/OsuAuthorize.php:1797]: Skipping the playcount check on the old site seems to only apply to users where the primary group is the bot group

Comment (nanaya) [reply app/Libraries/OsuAuthorize.php:1797]: the query is on `phpbb_user_group` 🤔

**Review by notbakaneko — APPROVED** [review]

**Review by notbakaneko — APPROVED** [review]

**Pull request merged**